helpers/spark-submit.py

```diff
@@ -16,55 +16,35 @@
 
     parser = argparse.ArgumentParser()
     parser.add_argument("--master", default=None, type=str, help='Kubernetes control plane uri.')
-    parser.add_argument("--properties-file", default=None, type=str, help='Spark default configuration properties file.')
-    parser.add_argument("--name", default="spark-app", type=str, help='Spark application\'s name.')
     parser.add_argument("--deploy-mode", default="client", type=str, help='Deployment mode for job submission. Default is \'client\'.')
-    parser.add_argument("--class", default=None, type=str, help='Java class of the application to be launched.')
+    parser.add_argument("--properties-file", default=None, type=str, help='Spark default configuration properties file.')
     parser.add_argument("--conf", action="append", type=str, help='Configuration properties to override the corresponding defaults.')
-    parser.add_argument("extra_args", nargs="*")
-    args = parser.parse_args()
+    args, extra_args = parser.parse_known_args()
 
     os.environ["HOME"] = pwd.getpwuid(os.getuid())[USER_HOME_DIR_ENT_IDX]
     if os.environ.get('SPARK_HOME') is None or os.environ.get('SPARK_HOME') == '':
         os.environ['SPARK_HOME'] = os.environ['SNAP']
 
-    submit_args = sys.argv[1:]
-    conf = dict()
-
-    if args.conf:
-        for c in args.conf:
-            try:
-                kv = c.split('=')
-                k = kv[0]
-                v = '='.join(kv[1:])
-                conf[k] = os.environ.get(v, v)
-            except IndexError as e:
-                logging.error('Configuration related arguments parsing error. Please check input arguments and try again.')
-                sys.exit(EXIT_CODE_BAD_CONF_ARG)
+    STATIC_DEFAULTS_CONF_FILE = utils.get_static_defaults_conf_file()
+    DYNAMIC_DEFAULTS_CONF_FILE = utils.get_dynamic_defaults_conf_file()
 
-    if not args.master:
-        USER_HOME_DIR = pwd.getpwuid(os.getuid())[USER_HOME_DIR_ENT_IDX]
-        DEFAULT_KUBECONFIG = f'{USER_HOME_DIR}/.kube/config'
-        kubeconfig = os.environ.get('KUBECONFIG') or DEFAULT_KUBECONFIG
-        kubectl_cmd = '{}/kubectl'.format(os.environ['SNAP'])
+    static_defaults = utils.read_property_file(STATIC_DEFAULTS_CONF_FILE)
+    dynamic_defaults = utils.read_property_file(DYNAMIC_DEFAULTS_CONF_FILE) if os.path.isfile(DYNAMIC_DEFAULTS_CONF_FILE) else dict()
+    user_defaults = utils.read_property_file(args.properties_file) if args.properties_file else dict()
+    user_overrides = utils.parse_conf_overrides(args.conf) if args.conf else dict()
```

Comment:
My idea was to provide the "--conf" flags straight into the submit command, with no parsing/creationg of the dict. But it is not a strong point